### PR TITLE
Implement IInstrument for DMM

### PR DESCRIPTION
**Pull request by gpachecoNI**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-labview/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

 - Creates an implementation of IInstrument for NI-DMM
 - Adds Tests for the Create / Attach / Detach and Close

Based on the NI-Digital

### Why should this Pull Request be merged?

Implements the following US [AB#2533859](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2533859)

### What testing has been done?

Auto Tests pass

### CODE REVIEW

**Review by jasonmreding — APPROVED** [review]

**Comment by dixonjoel** [comment]
There are 0 checks on this PR, presumably because of the branch conflicts.  Please ensure the checks run and the G tests pass before submitting. I'd like to do a last quick check after the conflicts are resolved, so you can mention me when that's done.

File: Source/Runtime/MeasurementLink niDMM/Instrument/Create.vi
Comment (gpachecoNI) [inline Source/Runtime/MeasurementLink niDMM/Instrument/Create.vi:1]: image

File: Source/Runtime/MeasurementLink niDMM/Instrument/Get Instrument Type ID.vi
Comment (gpachecoNI) [inline Source/Runtime/MeasurementLink niDMM/Instrument/Get Instrument Type ID.vi:1]: image

File: Source/Runtime/MeasurementLink niDMM/Instrument/Get Provided Interface.vi
Comment (gpachecoNI) [inline Source/Runtime/MeasurementLink niDMM/Instrument/Get Provided Interface.vi:1]: image

File: Source/Runtime/MeasurementLink niDMM/Instrument/Initialize MeasurementLink Session.vi
Comment (gpachecoNI) [inline Source/Runtime/MeasurementLink niDMM/Instrument/Initialize MeasurementLink Session.vi:1]: image

**Review by pbirkhol-ni — CHANGES_REQUESTED** [review]
Something is wrong with your project and/or libraries. I see this when I load the project.

image

Merge conflicts are hard in LabVIEW, even for text files. I have had the most luck by reverting all of my changes to projects and libraries. You can do this by removing your new library and tests from the project and saving all. Then merge in main and you shouldn't have any merge conflicts. Then re-add your new library and test VIs. This sounds silly, but it's fast and almost guaranteed to work. It won't work as well if you are modifying existing VIs, but that isn't the case here.

**Comment by gpachecoNI** [comment]
> use of the branch conflicts. Please ensure the checks run and the G tests pass before submitting. I'd like to do a last quick check after the conflicts are resolved, so y

@dixonjoel

**Review by pbirkhol-ni — APPROVED** [review]

**Comment by WesleyTangNationalInstruments** [comment]
Minor: There are some inconsistencies in the capitalization of "DMM" vs "Dmm" in folder names. For the tests, the Virtual Folder is "niDmm" while the disk folder is "niDMM".

measurement-services-labview\Source\Tests\Tests.Runtime\Measurement Server\niDMM

image

**Review by dixonjoel — APPROVED** [review]

**Pull request merged**